dapp/src/components/common/withBridge/index.jsx

```diff
@@ -23,96 +25,135 @@ export default function withBridge(WrappedComponent) {
     const foreignBridgeAddress = dashboard?.community?.foreignBridgeAddress
+    const foreignTokenAddress = dashboard?.community?.foreignTokenAddress
 
-    useInterval(() => {
-      if (bridge === 'foreign') {
-        fetchHomeBridgePastEvents({
-          networkType: foreignNetwork,
-          bridgeType,
-          transactionHash,
-          fromBlock: blockNumber,
-          homeBridgeAddress,
-          bridgeDirection,
-        }, { web3: web3Home }).then((events) => {
-          const bridgeEvent = getRelayEventByTransactionHash({ events, accountAddress })
-          if (bridgeEvent) {
-            setRelayEvent(bridgeEvent)
-            setDelay(null)
-            onConfirmation()
+    const fetchPastEvent = useCallback(() => {
+      fetchBridgePastEvents({
+        bridge,
+        networkType: foreignNetwork,
+        bridgeType,
+        bridgeDirection,
+        fromBlock: blockNumber,
+        homeBridgeAddress,
+        foreignBridgeAddress
+      }, {
+        web3: networkName !== 'fuse'
+          ? getWeb3({ networkType: 'fuse' })
+          : getWeb3({ networkType: foreignNetwork })
+      }).then((events) => {
+        const bridgeEvent = getRelayEventByTransactionHash({ events, accountAddress })
+        if (bridgeEvent) {
+          if (!foreignTokenAddress && bridge === 'foreign') {
+            if (get(bridgeEvent, 'returnValues.token')) {
+              dashboard.registerForeignTokenAddress({ foreignTokenAddress: get(bridgeEvent, 'returnValues.token') })
+              setNewTokenRegisteredDelay(null)
+            }
           }
-        })
-      } else {
-        fetchForeignBridgePastEvents({
-          networkType: foreignNetwork,
-          bridgeType,
-          fromBlock: blockNumber,
-          foreignBridgeAddress,
-          bridgeDirection,
-        }, { web3: web3Foreign }).then((events) => {
-          const bridgeEvent = getRelayEventByTransactionHash({ events, transactionHash, accountAddress })
-          if (bridgeEvent) {
-            setRelayEvent(bridgeEvent)
-            setDelay(null)
-            onConfirmation()
+          setDelay(null)
+          onConfirmation()
+        }
+      })
+    }, [
+      foreignNetwork,
+      bridgeType,
+      transactionHash,
+      blockNumber,
+      homeBridgeAddress,
+      bridge,
+      bridgeDirection,
+      foreignBridgeAddress,
+      foreignTokenAddress,
+      networkName,
+      dashboard?.bridgeStatus
+    ])
+
+    const fetchNewTokenRegistered = useCallback(() => {
+      fetchHomeNewTokenRegistered({
+        bridgeType,
+        bridgeDirection,
+        fromBlock: blockNumber
+      }, {
+        web3: getWeb3({ networkType: foreignNetwork })
+      }).then((events) => {
+        const newTokenRegisteredRelayEvent = getRelayEventByTransactionHash({ events, transactionHash, accountAddress })
+        if (newTokenRegisteredRelayEvent) {
+          if (get(newTokenRegisteredRelayEvent, 'returnValues.homeToken')) {
```

Comment:
Can you disable the interval if the is already created? In that case the tx is confirmed but the NewTokenRegistered event is not found